Preparation/MyStack/MyStackInter.cs

```diff
@@ -1,34 +1,27 @@
 using System;
 using System.Collections;
+using System.Data;
 
 namespace MyStack
 {
-    public class MyStack
+    public class MyStack : IEnumerable
     {
-        public static readonly int maxsize = 10;
+        public static int count = 1;
         public int topstack;
-        public int[] arrayint = new int[maxsize];
+        public int[] arrayint = new int[count];
 
-        public MyStack()
+        public bool Push(int add)
         {
-            topstack = 0;
-        }
-
-        public bool Add(int add)
-        {
-            if (topstack >= maxsize)
-            {
-                Console.WriteLine("Stack Overfllow");
-                return false;
-            }
-            else
+            if (count >= arrayint.Length)
```

Comment:
The condition will always be true, as you're every time resizing by 1. 
Such implementation is much closer to what is expected, but don't you think that reallocating memory every time just because of 1 item added is too memory inefficient?
How about initializing `arrayint` with 4 items, and every time the `arrayint.Length` exceeds `count`, do resize with `arrayint.Length * 2` new length.